lib/classes/webserver/class.DomainSSL.php

```diff
@@ -89,6 +90,10 @@ public function setDomainSSLFilesArray(array &$domain = null) {
 					$ssl_files['ssl_cert_chainfile'] = makeCorrectFile($sslcertpath.'/'.$domain['domain'].'_chain.pem');
 				}
 			}
+            		if (($dom_certs['ssl_ca_file'] != '') and ($dom_certs['ssl_cert_chainfile'] != '')) {
```

Comment:
and what if only one of these is set? What if another services config relies on the file **$sslcertpath.'/'.$domain['domain'].'_fullchain.pem'** to exist? The fullchain does not necessarily require both, ca and chainfile to be filled....with one of them, it's still the fullchain

Reply:
OK, i´m not that save with specification of fullchain certs, but can easily change the condition to `or` and check only existing one of them.
Apart from combining them manually: Isn´t fullchain not already retrieved from LE already but not stored in a file?
https://github.com/Froxlor/Froxlor/blob/4d3fa6eca510329542ec4518c4bc029533051d01/lib/classes/ssl/class.lescript.php#L331